pkg/registry/core/resourcequota/strategy_test.go

```diff
@@ -58,3 +60,84 @@ func TestResourceQuotaStrategy(t *testing.T) {
+			wantWarnings: []string{},
+		},
+		{
+			name: "Request less than limit",
+			args: &api.ResourceQuota{
+				Spec: api.ResourceQuotaSpec{
+					Hard: api.ResourceList{
+						api.ResourceName("requests.cpu"):               resource.MustParse("500m"),
+						api.ResourceName("limits.cpu"):                 resource.MustParse("1"),
+						api.ResourceName("requests.memory"):            resource.MustParse("1Gi"),
+						api.ResourceName("limits.memory"):              resource.MustParse("2Gi"),
+						api.ResourceName("requests.storage"):           resource.MustParse("1Gi"),
+						api.ResourceName("limits.storage"):             resource.MustParse("2Gi"),
+						api.ResourceName("requests.ephemeral-storage"): resource.MustParse("1Gi"),
+						api.ResourceName("limits.ephemeral-storage"):   resource.MustParse("2Gi"),
+					},
+				},
+			},
+			wantWarnings: []string{},
+		},
+		{
+			name: "Request greater than limit",
+			args: &api.ResourceQuota{
+				Spec: api.ResourceQuotaSpec{
+					Hard: api.ResourceList{
+						api.ResourceName("requests.cpu"):               resource.MustParse("2"),
+						api.ResourceName("limits.cpu"):                 resource.MustParse("1"),
+						api.ResourceName("requests.memory"):            resource.MustParse("3Gi"),
+						api.ResourceName("limits.memory"):              resource.MustParse("2Gi"),
+						api.ResourceName("requests.storage"):           resource.MustParse("3Gi"),
+						api.ResourceName("limits.storage"):             resource.MustParse("2Gi"),
+						api.ResourceName("requests.ephemeral-storage"): resource.MustParse("3Gi"),
+						api.ResourceName("limits.ephemeral-storage"):   resource.MustParse("2Gi"),
+					},
+				},
+			},
+			wantWarnings: []string{
+				"ResourceQuota requests.cpu (2) should be less than limits.cpu (1)",
+				"ResourceQuota requests.memory (3Gi) should be less than limits.memory (2Gi)",
+				"ResourceQuota requests.storage (3Gi) should be less than limits.storage (2Gi)",
+				"ResourceQuota requests.ephemeral-storage (3Gi) should be less than limits.ephemeral-storage (2Gi)",
+			},
+		},
+		{
+			name: "Request greater than limit, bare names",
+			args: &api.ResourceQuota{
+				Spec: api.ResourceQuotaSpec{
+					Hard: api.ResourceList{
+						api.ResourceName("cpu"):           resource.MustParse("2"),
+						api.ResourceName("limits.cpu"):    resource.MustParse("1"),
+						api.ResourceName("memory"):        resource.MustParse("3Gi"),
+						api.ResourceName("limits.memory"): resource.MustParse("2Gi"),
+					},
+				},
+			},
+			wantWarnings: []string{
+				"ResourceQuota cpu (2) should be less than limits.cpu (1)",
+				"ResourceQuota memory (3Gi) should be less than limits.memory (2Gi)",
+			},
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			warnings := Strategy.WarningsOnCreate(context.Background(), tt.args)
+			if len(warnings)+len(tt.wantWarnings) > 0 && !reflect.DeepEqual(warnings, tt.wantWarnings) {
```

Comment:
just a note, it is always better to sort the arrays before using deepequal to avoid flakes , but in this case it seems the order is guarantee because of how the function parses the resources